### ISSUE TITLE
Please wait until the caching process is finished

### ISSUE THREAD

**Issue by jaybihola** [jopemachine/alfred-evernote-workflow#70]
Every time I try to use ens to search for notes, I receive the message "Please wait until the caching process is finished." It has been several days since I have installed the workflow.

I have tried redoing the setup process but still, I am getting the same error.

I am running Mac OS Big Sur Beta on a Macbook Pro 15 inch.

**Comment by jopemachine** [comment]
Could you try to follow the below instructions?

```
1. en > clearcache
2. en > savecache, and wait until an notification shows up indicating caching is complete.
3. ens
```
This workflow is not tested on Big Sur beta yet.

If the same error continuing occurs, you can disable the caching feature (just use `en > clearcache`)

**Comment by jaybihola** [comment]
I did try the three steps. The issue is still happening. Just an additional note, I am using the new version of Evernote (Evernote 10).

**Comment by jopemachine** [comment]
Thanks for letting me know this.

It seems that Evernote 10 doesn't support AppleScript anymore.

In fact, I didn't know this fact because I didn't update it to Evernote 10 yet.

So, the search feature itself works,(because it use sdk) but the ability to open or cache notes does not work in Evernote 10.

(and all functions that need to AppleScript won't working.)

I need to think about if this is feasible to implement these features without going through Apple Script.

**Comment by jaybihola** [comment]
Okay, for now, I will downgrade to an older version of Evernote.

**Comment by PaulAmosKreiner** [comment]
I am having the same issue outside of version 10.

I am using the Evernote Legacy 7.14.1 version (which I downloaded after updating to version 10 and realizing that it sucks in so many ways).

I have let it run for ~1 hour since my evernote account does have quite a lot of content and gave him one retry with clearcache followed by savecache – still nothing.

is there anything else i can try?

running 10.15.7 (catalina).

EDIT: creating notes via the workflow works so there is no fundamental disconnect between the workflow and evernote.

**Comment by 8988cowdyj** [comment]
> I am having the same issue outside of version 10.
> 
> I am using the Evernote Legacy 7.14.1 version (which I downloaded after updating to version 10 and realizing that it sucks in so many ways).
> 
> I have let it run for ~1 hour since my evernote account does have quite a lot of content and gave him one retry with clearcache followed by savecache – still nothing.
> 
> is there anything else i can try?
> 
> running 10.15.7 (catalina).
> 
> EDIT: creating notes via the workflow works so there is no fundamental disconnect between the workflow and evernote.

Just installed and got this running today. I had to modify the saveCache.scpt file to reference Evernote Legacy instead of "Evernote" since the Legacy app has its own name.

**Comment by jopemachine** [comment]
> I am having the same issue outside of version 10.
> 
> I am using the Evernote Legacy 7.14.1 version (which I downloaded after updating to version 10 and realizing that it sucks in so many ways).
> 
> I have let it run for ~1 hour since my evernote account does have quite a lot of content and gave him one retry with clearcache followed by savecache – still nothing.
> 
> is there anything else i can try?
> 
> running 10.15.7 (catalina).
> 
> EDIT: creating notes via the workflow works so there is no fundamental disconnect between the workflow and evernote.

Yes, I've received some emails saying there is a bug in caching process.
I think, unfortunately, this bug depends on the account and account's note.
(So I have had difficulty to debug it because I cannot reproduce it..)
I think there might be some possibilities, but currently, Evernote has been updated to 10 and I've stopped debugging.

**Comment by jopemachine** [comment]
> > I am having the same issue outside of version 10.
> > I am using the Evernote Legacy 7.14.1 version (which I downloaded after updating to version 10 and realizing that it sucks in so many ways).
> > I have let it run for ~1 hour since my evernote account does have quite a lot of content and gave him one retry with clearcache followed by savecache – still nothing.
> > is there anything else i can try?
> > running 10.15.7 (catalina).
> > EDIT: creating notes via the workflow works so there is no fundamental disconnect between the workflow and evernote.
> 
> Just installed and got this running today. I had to modify the saveCache.scpt file to reference Evernote Legacy instead of "Evernote" since the Legacy app has its own name.

Thanks for letting me know.
I would like to reflect your point if is possible.
May I ask what changes you have to make in saveCache.scpt with Legacy download version? 
e.g. com.evernote.Evernote => ?

And May I ask where you could download evernote 7 version, too?
I've founded evernote 7.14 on some url but it says its version is too old and I can't open it.

**Comment by 8988cowdyj** [comment]
> > > I am having the same issue outside of version 10.
> > > I am using the Evernote Legacy 7.14.1 version (which I downloaded after updating to version 10 and realizing that it sucks in so many ways).
> > > I have let it run for ~1 hour since my evernote account does have quite a lot of content and gave him one retry with clearcache followed by savecache – still nothing.
> > > is there anything else i can try?
> > > running 10.15.7 (catalina).
> > > EDIT: creating notes via the workflow works so there is no fundamental disconnect between the workflow and evernote.
> > 
> > 
> > Just installed and got this running today. I had to modify the saveCache.scpt file to reference Evernote Legacy instead of "Evernote" since the Legacy app has its own name.
> 
> Thanks for letting me know.
> I would like to reflect your point if is possible.
> May I ask what changes you have to make in saveCache.scpt with Legacy download version?
> e.g. com.evernote.Evernote => ?
> 
> And May I ask where you could download evernote 7 version, too?
> I've founded evernote 7.14 on some url but it says its version is too old and I can't open it.

You can download an older version of the app (which Evernote is calling "Evernote Legacy") from their [Help & Learning page](https://help.evernote.com/hc/en-us/articles/360052560314).

I renamed the app "Evernote_Legacy" and then changed this line of saveCache.script: 
     from:
          `do shell script "open /Applications/Evernote.app"`
     to:
          `do shell script "open /Applications/Evernote_Legacy.app"`

I didn't need to change the application id section of the code.

Hope that helps!

**Comment by PaulAmosKreiner** [comment]
Can confirm the solution of @8988cowdyj ! However, having updated to Big Sur in the meantime, I am now encountering additional problems.

What works now: I managed to manually run the saveCache.scpt based on @8988cowdyj 's hint. the search_content folder now exists and looks correct.

What does not work: I can't use any of the commands that work based on a script filter, e.g. ens. The debugger points me to this:

`
16:57:33.053] Logging Started...
[16:57:36.026] alfred-evernote-workflow[Script Filter] Queuing argument ''
[16:57:36.093] alfred-evernote-workflow[Script Filter] Script with argv '(null)' finished
[16:57:36.094] ERROR: alfred-evernote-workflow[Script Filter] Code 127: /bin/bash: ./node_modules/.bin/run-node: No such file or directory
`

(my input was just "ens".)

node_modules sits in its default place (home) and contains the ".bin" folder which itself contains the run-node alias that the script tries to open.

EDIT: fixed it. just manually changed the path in the script filter to be relative to home. was relative to working directory – and i don't know what that is at the execution there. now it all works, however I had to manually change all the script filter paths... anyone knows what happened here?

**Comment by imbutter** [comment]
I have the same error.

> My MacOS version is 10.14.6 
> Alferd is 4.6.4
> Workflow is 0.9.7
> Evernote is Legacy 7.14.1

the 1st error is :type 'en' can not open the evernote legacy
the 2nd error: cannot use cache , I tried the same steps (clear-> save) over 3 times, but it still can not work
if the workflow 0.9.8 can solve this problem? but I can not update it with npm, please update it to packal.org?
or I can do this change myself 
```
I renamed the app "Evernote_Legacy" and then changed this line of saveCache.script:
from:
do shell script "open /Applications/Evernote.app"
to:
do shell script "open /Applications/Evernote_Legacy.app"
```

**Comment by imbutter** [comment]
I changed the saveCache.script.so it works!!!
